client/src/components/WorkflowInvocationState/WorkflowInvocationOverview.vue

```diff
@@ -108,8 +108,10 @@ const invocationPdfLink = computed<string | null>(() => {
     }
 });
 
-const hasMessages = computed<boolean>(() => {
-    return props.invocation?.messages.length ? true : false;
+const uniqueMessages = computed(() => {
+    const messages = props.invocation?.messages || [];
+    const uniqueMessagesSet = new Set(messages.map((message) => JSON.stringify(message)));
+    return Array.from(uniqueMessagesSet).map((message) => JSON.parse(message)) as InvocationMessageResponseModel[];
```

Comment:
The manual type assignment here seems odd and likely to drift from the source defined in the API in the future. props.invocation.messages should already have the correct type, so type inference should also reveal the correct type on the set -> array conversion.